datalad_next/iter_collections/annexworktree.py

```diff
@@ -0,0 +1,208 @@
+"""Report on the content of a Git-annex repository worktree
+"""
+from __future__ import annotations
+
+import logging
+from dataclasses import dataclass
+from pathlib import (
+    Path,
+    PurePath,
+    PurePosixPath,
+)
+from typing import (
+    Any,
+    Generator,
+)
+
+from more_itertools import intersperse
+
+from .gitworktree import (
+    GitWorktreeItem,
+    GitWorktreeFileSystemItem,
+    FileSystemItemType,
+    iter_gitworktree
+)
+from datalad_next.itertools import (
+    itemize,
+    load_json,
+    route_in,
+    route_out,
+    StoreOnly,
+)
+from datalad_next.runners import iter_subproc
+
+
+lgr = logging.getLogger('datalad.ext.next.iter_collections.annexworktree')
+
+
+@dataclass
+class AnnexWorktreeItem(GitWorktreeItem):
+    annexkey: str | None = None
+    annexsize: int | None = None
+    annexobjpath: PurePath | None = None
+
+
+@dataclass
+class AnnexWorktreeFileSystemItem(GitWorktreeFileSystemItem):
+    annexkey: str | None = None
+    annexsize: int | None = None
+    annexobjpath: PurePath | None = None
+
+
+def content_path(item: AnnexWorktreeItem | AnnexWorktreeFileSystemItem,
+                 base_path: Path,
+                 link_target: bool,
+                 ) -> PurePath | None:
+
+    if item.annexobjpath:
+        return \
+            item.annexobjpath \
+            if (base_path / item.annexobjpath).exists() \
+            else None
+
+    # if this is not an annexed file, honor `link_target`-parameter.
+    if item.type == FileSystemItemType.file:
+        return item.name
+    elif item.type == FileSystemItemType.symlink:
+        return item.link_target if link_target is True else None
+    return item.link_target if link_target else item.name
+
+
+def get_annex_item(data, **kwargs):
```

Comment:
I think it would help to rename this to something like `get_worktree_item`. it is true that and `Annex***Item` comes back, but one can have the misconception that any item refers to an annexed file. This is not the case.

Reply:
That is a good point, I will change it.